### ISSUE TITLE
error logging in as doctor while currently signed in with user

### ISSUE THREAD

**Issue by vulong264** [LTKFoundation/clinic_booking_project#19]
![devise login problem](https://cloud.githubusercontent.com/assets/8022714/20479919/be033946-b012-11e6-83d1-a9f27c5da83c.gif)

Guys @kiendt07 @tucq88 @taduyvinh . I'm logging in to book as a user. When I'm done. I click on the doctor signin at the bottom, then login with Doctor account. Login succeed but I'm still logon as User. Can you have a look?

**Comment by kiendt07** [comment]
@vulong264 that's because in application.html.erb we check `if current_user` then there is a hmtl code, `if current_doctor` then there is another html code. And because of 2 devise model, both current_user and current_doctor return an object when you logged in as user and then logged as doctor

**Comment by vulong264** [comment]
and how are we going to fix that @kiendt07 ? You have a solution already?

**Comment by kiendt07** [comment]
no i think that's another problem of 2 devise model. current_user always returns an object when u r not logged out yet.

**Comment by vulong264** [comment]
so we are not going to do anything about it? are you serious @kiendt07 ??
Just a thought, can you solve it by force logout when a usertype clicked on signin link of another usertype? Better yet, force logout the existing session when user input a correct password for another usertype? Can you do that?

**Comment by kiendt07** [comment]
yeah im thinking about the solution. that must be override 2 devise sessions_controller. Let me do that

**Comment by vulong264** [comment]
just commented on slack already. If it's too much work. Just hide or disable the link "Doctor signin". Or maybe when user click on "Doctor signin", you check if they are currently logged on as a Client, then you show a page saying that you must log out before log in as Doctor. Then proceed. How about that?

**Comment by taduyvinh** [comment]
Just like you said, to make it quick just use user_sign_in? and doctor_sign_in? for each <session/new.html.erb>  to prevent users and doctors from logging in as different roles. New sessions controller is just too much work i think

**Comment by kiendt07** [comment]
> Hide doctor's authentication link when logged in as user
@vulong264 can i close this now?

**Comment by tucq88** [comment]
@kiendt07 Don't hide the link, keep the link but when open Doctor login form, instead of display form - we'll show an notice message asking user to log out first ? Is it better UX ?
